src/session/client-certificate.c

```diff
@@ -328,8 +377,37 @@ cockpit_session_client_certificate_map_user (const char *client_certificate_file
       return NULL;
     }
 
+  /* check if stdin is a Unix socket; then we are being called via cockpit-session@.service and need to
+   * check the cgroup of our peer */
+  struct ucred ucred;
+  int ws_cgroup_fd;
+  if (getsockopt (STDIN_FILENO, SOL_SOCKET, SO_PEERCRED, &ucred, &(socklen_t){sizeof ucred}) == 0)
+    {
+      debug ("unix socket mode, reading cgroup from peer pid %d", ucred.pid);
+      ws_cgroup_fd = open_proc_pid (ucred.pid);
```

Comment:
This name is highly misleading: it doesn't have anything to do with cgroups (yet).  It's an fd to the `/proc/` entry for our peer process.

Reply:
Renamed to ws_pid_dirfd, similarly my_pid_dirfd.